overlord/state/taskrunner.go

```diff
@@ -302,6 +326,22 @@ func (r *TaskRunner) Ensure() {
 	}
 }
 
+func (r *TaskRunner) cleanTask(t *Task) {
+	select {
+	case <-t.Change().Ready():
+	default:
+		return
+	}
+	cleanup, ok := r.cleanups[t.Kind()]
+	if ok {
+		if err := cleanup(t); err != nil {
```

Comment:
if I understand this is synchronous and done with the State lock held? do we assume they are fast? or do we need to setup goroutines for them like the other handlers, both ways maybe AddCleanup should grow a bit more description about these conditions they are run under